packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -95,6 +100,32 @@ class TextPainter {
     _needsLayout = true;
   }
 
+  /// An optional maximum number of lines for the text to span, wrapping if necessary.
+  /// If the text exceeds the given number of lines, it will be truncated according
+  /// to [overflow].
+  int get maxLines => _maxLines;
+  int _maxLines;
+  set maxLines(int value) {
+    assert(value == null);
+    if (_maxLines == value)
+      return;
+    _maxLines = value;
+    _paragraph = null;
+    _needsLayout = true;
+  }
```

Comment:
Why is this a property on the text painter?  I would have expected it to come in through the TextStyle only